core-models/src/main/scala/com/mesosphere/usi/core/models/faultdomain/RegionFilter.scala

```diff
@@ -5,4 +5,6 @@ case class RegionFilter(region: String) extends DomainFilter {
   override def apply(masterDomain: Protos.DomainInfo, nodeDomain: Protos.DomainInfo): Boolean = {
```

Comment:
What do you think about a `MatchResult` type?

Reply:
Hmm, maybe?  I was hoping it'd be enough detail to have the descriptions of the filters that didn't match:

```
Offer did not match; 1 filter didn't match: "expect region 'windows' 
```

For the offer match debugging result that's processed by the implementation framework, the actual non-matching filters and matching filters will be passed